### ISSUE TITLE
Add support for JDK 25

### ISSUE THREAD

**Issue by goldmann** [nvuillam/npm-groovy-lint#610]
**Description**

`npm-groovy-lint` auto-downloads a JRE via `java-caller` when none is found. The supported range is hardcoded to 17–24, so Java 25 triggers a broken retry loop that leaves stale files in `/tmp/njre/` and fails with different errors on each attempt (`TAR_BAD_ARCHIVE` → `ENOENT` → `EEXIST`).

**Error**

```
02:28:56  Java jre or jdk between 17 and 24 is required 
02:28:56  Installing Java jre 24 in /home/jenkins/.java-caller...
02:28:56  Unexpected error: TAR_BAD_ARCHIVE: Unrecognized archive format
02:28:56  Error: TAR_BAD_ARCHIVE: Unrecognized archive format
02:28:56      at za (/home/jenkins/.cache/pre-commit/repohzy1hop2/node_env-system/lib/node_modules/npm-groovy-lint/node_modules/tar/dist/commonjs/index.min.js:3:3159)
02:28:56      at Yi.warn (/home/jenkins/.cache/pre-commit/repohzy1hop2/node_env-system/lib/node_modules/npm-groovy-lint/node_modules/tar/dist/commonjs/index.min.js:3:5483)
02:28:56      at Yi.warn (/home/jenkins/.cache/pre-commit/repohzy1hop2/node_env-system/lib/node_modules/npm-groovy-lint/node_modules/tar/dist/commonjs/index.min.js:3:50730)
02:28:56      at Yi.<anonymous> (/home/jenkins/.cache/pre-commit/repohzy1hop2/node_env-system/lib/node_modules/npm-groovy-lint/node_modules/tar/dist/commonjs/index.min.js:3:4649)
02:28:56      at Yi.emit (node:events:530:35)
02:28:56      at [emit] (/home/jenkins/.cache/pre-commit/repohzy1hop2/node_env-system/lib/node_modules/npm-groovy-lint/node_modules/tar/dist/commonjs/index.min.js:3:7579)
02:28:56      at [maybeEnd] (/home/jenkins/.cache/pre-commit/repohzy1hop2/node_env-system/lib/node_modules/npm-groovy-lint/node_modules/tar/dist/commonjs/index.min.js:3:10206)
02:28:56      at Yi.end (/home/jenkins/.cache/pre-commit/repohzy1hop2/node_env-system/lib/node_modules/npm-groovy-lint/node_modules/tar/dist/commonjs/index.min.js:3:11217)
02:28:56      at Gt.end (/home/jenkins/.cache/pre-commit/repohzy1hop2/node_env-system/lib/node_modules/npm-groovy-lint/node_modules/tar/dist/commonjs/index.min.js:1:2062)
02:28:56      at [emitEnd2] (/home/jenkins/.cache/pre-commit/repohzy1hop2/node_env-system/lib/node_modules/npm-groovy-lint/node_modules/tar/dist/commonjs/index.min.js:1:8393)
02:28:56  Java jre or jdk between 17 and 24 is required 
02:28:56  Installing Java jre 24 in /home/jenkins/.java-caller...
02:28:56  Unexpected error: ENOENT: no such file or directory, copyfile '/tmp/njre/OpenJDK24U-jre_x64_linux_hotspot_24.0.2_12.tar.gz' -> '/home/jenkins/.java-caller/OpenJDK24U-jre_x64_linux_hotspot_24.0.2_12.tar.gz'
02:28:56  Error: ENOENT: no such file or directory, copyfile '/tmp/njre/OpenJDK24U-jre_x64_linux_hotspot_24.0.2_12.tar.gz' -> '/home/jenkins/.java-caller/OpenJDK24U-jre_x64_linux_hotspot_24.0.2_12.tar.gz'
02:28:56  Java jre or jdk between 17 and 24 is required 
02:28:56  Installing Java jre 24 in /home/jenkins/.java-caller...
02:28:56  Unexpected error: ENOENT: no such file or directory, unlink '/tmp/njre/OpenJDK24U-jre_x64_linux_hotspot_24.0.2_12.tar.gz'
02:28:56  Error: ENOENT: no such file or directory, unlink '/tmp/njre/OpenJDK24U-jre_x64_linux_hotspot_24.0.2_12.tar.gz'
02:28:56  Java jre or jdk between 17 and 24 is required 
02:28:56  Installing Java jre 24 in /home/jenkins/.java-caller...
02:28:56  Unexpected error: ENOENT: no such file or directory, unlink '/tmp/njre/OpenJDK24U-jre_x64_linux_hotspot_24.0.2_12.tar.gz'
02:28:56  Error: ENOENT: no such file or directory, unlink '/tmp/njre/OpenJDK24U-jre_x64_linux_hotspot_24.0.2_12.tar.gz'
02:28:56  Java jre or jdk between 17 and 24 is required 
02:28:56  Installing Java jre 24 in /home/jenkins/.java-caller...
02:28:56  Unexpected error: EEXIST: file already exists, mkdir '/tmp/njre'
02:28:56  Error: EEXIST: file already exists, mkdir '/tmp/njre'
```

**Fix needed**

Bump the supported JRE ceiling to include Java 25+.
